packages/styles/settings-tools/_s.inline-icons.scss

```diff
@@ -1,59 +1,121 @@
+@function str-replace($string, $search, $replace: "") {
+  $index: str-index($string, $search);
+
+  @if $index {
+    @return str-slice($string, 1, $index - 1) + $replace + str-replace(
+      str-slice($string, $index + str-length($search)),
+      $search,
+      $replace
+    );
+  }
+
+  @return $string;
+}
+
+$encoding-reference: (
+  ("%", "%25"),
+  // Encode "%" first, otherwise the "%" from encoded code would be encoded again (which would be bad)
+  ("<", "%3C"),
+  (">", "%3E"),
+  ("'", "%22"),
+  // Replace " with ' because that's shorter than %22 and normally working
+  ('"', "'"),
+  ("#", "%23"),
+  ("&", "%26")
+);
+
+@function svg-encode($svg) {
+  @each $char, $encoded in $encoding-reference {
+    $svg: str-replace($svg, $char, $encoded);
+  }
+
+  @return "data:image/svg+xml," + $svg;
+}
+
 /* stylelint-disable max-line-length */
 @function inline-icons($icon, $fill) {
```

Comment:
Great job @mohamedMok ! 🔥 🔥 🔥 🔥 

Do you think we could make the code of this function a little less verbose, by making only one return at the end of the function?

Something like :

```SASS
@function inline-icons($icon, $fill) {
  $svg: '';

  @if $icon == "cross-16" {
    $svg: '<svg xmlns="http://www.w3.org/2000/svg" height="1rem" width="1rem" fill="#{$fill}" viewBox="0 0 16 16"><path d="M9.41 8l3.3-3.29a1 1 0 10-1.42-1.42L8 6.59l-3.29-3.3a1 1 0 00-1.42 1.42L6.59 8l-3.3 3.29a1 1 0 000 1.42 1 1 0 001.42 0L8 9.41l3.29 3.3a1 1 0 001.42 0 1 1 0 000-1.42z"/></svg>';
  }
  
  // ... 
  
  @return svg-encode($svg);
}
```